### PR TITLE
Add the get and get_mut functions

### PR DESCRIPTION
**Pull request by C4K3**
The functions are analogous to Vec's get and get_mut. index and
index_mut will panic if you try to access something out of bounds,
requiring users of the library to manually do bounds checking in their
code. By adding these functions this will no longer be necessary.

### CODE REVIEW

**Comment by tailhook** [comment]
Looks good. But I'm not sure why code for two functions is so much different? It should look like very similar for mutable and immutable case.

File: src/buf.rs [hunk 1]
```diff
@@ -391,6 +391,32 @@ impl Buf {
         }
         tail
     }
+
+    /// Returns the element at the index, or `None` if the index is
```
Comment (tailhook) [inline src/buf.rs]: We should say `byte` instead of `element`. We don't operate anything except bytes.

**Comment by C4K3** [comment]
The reason they're different is that I wrote the immutable one first, but then had to do it differently for the mutable one because of the borrow checker, but you're absolutely right that it is nicer if they're almost identical.

Fixed that and changed element into byte.

**Comment by tailhook** [comment]
Merged. Thanks!

**Pull request merged**